src/post.ts

```diff
@@ -1,20 +1,27 @@
 import {EventPayload} from './EventPayload';
 
 /**
- * Used by the AnalyticsEventReport report() method to send an analytics event request via the
- * Beacon API.
- *
- * todo: Update this method to default to using fetch + keepalive unless not supported and
- * forceFetch is not set to true. Also update the description above.
+ * Used by the AnalyticsEventReport report() method to send an analytics event request via
+ * fetch, with the keepalive option set to true. If forceFetch is false and keepalive is not
+ * supported by the browser i.e. Firefox, the request is sent via the Beacon API.
  *
  * @param url URL that the request will be sent to
  * @param body the EventPayload object
+ * @param forceFetch if true, it will force the browser to call fetch instead of sendBeacon,
+ * regardless of whether keepAlive is supported
  */
-const post = (
```

Comment:
Thoughts on making a helper function which returns a boolean to use beacon or fetch? 

Then can unit test that based on the user agent etc

Reply:
Sounds good, I think that would be cleaner. I added a helper function in a new file (browser.ts).